src/test.rs

```diff
@@ -24,47 +24,36 @@ macro_rules! promote_and_back {
-    u16   => f32, f64,          i32, isize, i64,          u32, usize, u64;
-    u32   => f32, f64,                      i64,                      u64;
-    usize => f32, f64,                      i64,                      u64;
-    u64   => f32, f64                                                    ;
+    i8    => f32, f64,     i16, i32, isize, i64, i128                            ;
+    i16   => f32, f64,          i32, isize, i64, i128                            ;
+    i32   => f32, f64,                      i64, i128                            ;
+    isize => f32, f64,                      i64, i128                            ;
+    i64   => f32, f64,                           i128                            ;
+    u8    => f32, f64,     i16, i32, isize, i64, i128, u16, u32, usize, u64, u128;
+    u16   => f32, f64,          i32, isize, i64, i128,      u32, usize, u64, u128;
+    u32   => f32, f64,                      i64, i128,                  u64, u128;
+    usize => f32, f64,                      i64, i128,                  u64, u128;
+    u64   => f32, f64,                           i128,                       u128;
 }
 
 #[cfg(target_pointer_width = "64")]
 promote_and_back! {
-    i8    => f32, f64,     i16, i32, i64, isize                          ;
-    i16   => f32, f64,          i32, i64, isize                          ;
-    i32   => f32, f64,               i64, isize                          ;
-    i64   => f32, f64                                                    ;
-    isize => f32, f64                                                    ;
-    u8    => f32, f64,     i16, i32, i64, isize,     u16, u32, u64, usize;
-    u16   => f32, f64,          i32, i64, isize,          u32, u64, usize;
-    u32   => f32, f64,               i64, isize,               u64, usize;
-    u64   => f32, f64                                                    ;
-    usize => f32, f64                                                    ;
+    i8    => f32, f64,     i16, i32, i64, isize, i128                            ;
+    i16   => f32, f64,          i32, i64, isize, i128                            ;
+    i32   => f32, f64,               i64, isize, i128                            ;
+    i64   => f32, f64,                           i128                            ;
+    isize => f32, f64,                           i128                            ;
+    u8    => f32, f64,     i16, i32, i64, isize, i128, u16, u32, u64, usize, u128;
+    u16   => f32, f64,          i32, i64, isize, i128,      u32, u64, usize, u128;
+    u32   => f32, f64,               i64, isize, i128,           u64, usize, u128;
+    u64   => f32, f64,                           i128,                       u128;
+    usize => f32, f64,                           i128,                       u128;
 }
 
-// TODO uncomment this once quickcheck supports Arbitrary for i128/u128
-// https://github.com/BurntSushi/quickcheck/issues/162
-/*#[cfg(feature = "x128")]
+// TODO uncomment this.
+/*
```

Comment:
The `Arbitrary` support for u128 and i128 seems to work, but I couldn't figure out how to enable these two lines, so I left them commented out.

```
i128 => f32, f64 ;
u128 => f32, f64 ;
```